### PR TITLE
UCP/UCT/TAG: Remove TM_OFFLOAD var

### PR DESCRIPTION
**Pull request by brminich**
- UCP/TAG: Remove TM_OFFLOAD env var
- UCP/TAG: Use AM score func for tag lane
- UCP/TAG: Tag lane can be a proxy lane
- UCT/RC: Disable TM by default
- TOOLS/INFO: Fix ucp_ep_create error message

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5975/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5990/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@hoopoepg, pls review

File: src/ucp/wireup/select.c [hunk 1]
```diff
@@ -1011,11 +944,25 @@ static ucs_status_t ucp_wireup_add_tag_lane(ucp_ep_h ep, unsigned address_count,
...
+    criteria.local_md_flags     = UCT_MD_FLAG_REG; /* needed for posting tags to HW */
+    criteria.remote_md_flags    = UCT_MD_FLAG_REG; /* needed for posting tags to HW */
+    criteria.remote_iface_flags = /* the same as local_iface_flags */
+    criteria.local_iface_flags  = UCT_IFACE_FLAG_TAG_EAGER_BCOPY |
+                                  UCT_IFACE_FLAG_TAG_RNDV_ZCOPY  |
+                                  UCT_IFACE_FLAG_GET_ZCOPY       |
+                                  UCT_IFACE_FLAG_PENDING;
+    criteria.calc_score         = ucp_wireup_am_score_func;
+    criteria.tl_rsc_flags       = 0;
+
```
Comment (hoopoepg) [inline src/ucp/wireup/select.c]: should proxy lane be configured?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5999/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6007/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6009/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@hoopoepg, fixed

**Comment by brminich** [comment]
@yosefe, plz check

**Pull request merged**